django/db/models/functions/math.py

```diff
@@ -0,0 +1,117 @@
+from django.db.models import Func, Transform, fields
+
+
+class Abs(Transform):
+    function = 'ABS'
+    lookup_name = 'abs'
+
+
+class Acos(Transform):
+    function = 'ACOS'
+    lookup_name = 'acos'
+    output_field = fields.FloatField()
+
+
+class Asin(Transform):
+    function = 'ASIN'
+    lookup_name = 'asin'
+    output_field = fields.FloatField()
+
+
+class Atan(Transform):
+    function = 'ATAN'
+    lookup_name = 'atan'
+    output_field = fields.FloatField()
+
+
+class Atan2(Func):
+    function = 'ATAN2'
+    arity = 2
+    output_field = fields.FloatField()
+
+
+class Ceiling(Transform):
+    function = 'CEILING'
+    lookup_name = 'ceiling'
+
+
+class Cos(Transform):
+    function = 'COS'
+    lookup_name = 'cos'
+    output_field = fields.FloatField()
+
+
+class Cot(Transform):
+    function = 'COT'
+    lookup_name = 'cot'
+    output_field = fields.FloatField()
+
+
+class Degrees(Transform):
+    function = 'DEGREES'
+    lookup_name = 'degrees'
+    output_field = fields.FloatField()
+
+
+class Exp(Transform):
+    function = 'EXP'
+    lookup_name = 'exp'
+
+
+class Floor(Transform):
+    function = 'FLOOR'
+    lookup_name = 'floor'
+
+
+class Log(Transform):
```

Comment:
I think that we should use `Log()` for `LOG(B, X)` and create `Ln()` for `LN(X)`. See the following for details:

```
PostgreSQL  MySQL           Oracle          SQLite
----------------------------------------------------------
LN(X)       LOG(X)/LN(X)    LN(x)            math.log(X)
---         LOG2(X)         ---              math.log2(X)
LOG(X)      LOG10(X)        ---              math.log10(X)
LOG(B, X)   LOG(B, X)       LOG(B, X)        math.log(X, B)

https://www.postgresql.org/docs/current/static/functions-math.html#FUNCTIONS-MATH-FUNC-TABLE
https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html
https://docs.oracle.com/en/database/oracle/oracle-database/12.2/sqlrf/LOG.html#GUID-3739F356-A4A0-4D0D-A4EB-9725ACA05CD1
https://docs.oracle.com/en/database/oracle/oracle-database/12.2/sqlrf/LN.html#GUID-DCC9EDAA-D308-4145-8E05-8D06A5EF5F6F
```

Reply:
I have created Ln and LOG functions. I noticed that different database behaves slightly differently. For example, in postgresql the input for LOG(B, X) should be numeric but not double precision, meaning that the input shouldn't be float. As a result, I removed the test_log.py using float numbers as input. Another strange thing is that although spatialite also has a LOG(B, X) with the first parameter to be base, when I ran a test, the result seems to be the reciprocal to what I expected. As a result, I modified the test_log.py accordingly.